### PR TITLE
AST: Make declLoc a LocOffsets rather than a Loc.

### PR DESCRIPTION
**Pull request by jvilk-stripe**
AST: Make declLoc a LocOffsets rather than a Loc. Now all locs in trees are not dependent on FileRef.

### Motivation

* Unblocks a Sorbet file hashing optimization that avoids re-parsing the file for hashing.
* Reduces LSP memory consumption a little bit (LSP holds all ASTs in memory).

### Test plan

Covered by existing tests.

### CODE REVIEW

File: ast/Trees.h [hunk 1]
```diff
@@ -341,12 +341,12 @@ TREE(ClassDef) : public Expression {
 
     void _sanityCheck();
 };
-CheckSize(ClassDef, 136, 8);
+CheckSize(ClassDef, 128, 8);
```
Comment (jez) [inline ast/Trees.h:344]: Nice

File: ast/Trees.h [hunk 2]
```diff
@@ -341,12 +341,12 @@ TREE(ClassDef) : public Expression {
...
 
 TREE(MethodDef) : public Expression {
 public:
     const core::LocOffsets loc;
-    core::Loc declLoc;
+    core::LocOffsets declLoc;
```
Comment (jvilk-stripe) [inline ast/Trees.h:349]: MethodDef's size didn't change (probably due to object padding)

**Review by jez — APPROVED** [review]
This is super rote, right? I'm not missing some gotcha?

**Comment by jvilk-stripe** [comment]
@jez Correct! The only non-rote thing is a simplification to `serialize.cc` -- some specialized routines for storing locs w/o files are no longer needed.

**Pull request merged**